src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -1241,6 +1199,90 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 
 		};
 
+	struct Volume
+	{
+		Volume( const IECore::CompoundObject *attributes )
+		{
+			volumeGrids = attribute<IECore::StringVectorData>( g_volumeGridsAttributeName, attributes );
+			velocityGrids = attribute<IECore::StringVectorData>( g_velocityGridsAttributeName, attributes );
+			velocityScale = attributeValue<float>( g_velocityScaleAttributeName, attributes, 1.0f );
+			velocityFPS = attributeValue<float>( g_velocityFPSAttributeName, attributes, 24.0f );
+			velocityOutlierThreshold = attributeValue<float>( g_velocityOutlierThresholdAttributeName, attributes, 0.001f );
+		}
+
+		IECore::ConstStringVectorDataPtr volumeGrids;
+		IECore::ConstStringVectorDataPtr velocityGrids;
+		float velocityScale;
+		float velocityFPS;
+		float velocityOutlierThreshold;
+
+		void hash( IECore::MurmurHash &h ) const
+		{
+			if( volumeGrids )
+			{
+				volumeGrids->hash( h );
+			}
+
+			if( velocityGrids )
+			{
+				velocityGrids->hash( h );
+			}
+
+			h.append( velocityScale );
+			h.append( velocityFPS );
+			h.append( velocityOutlierThreshold );
+		}
+
+		void apply( AtNode *node ) const
+		{
+			if( volumeGrids && volumeGrids->readable().size() )
+			{
+				AtArray *array = ParameterAlgo::dataToArray( volumeGrids.get(), AI_TYPE_STRING );
+				AiNodeSetArray( node, g_volumeGridsArnoldString, array );
+			}
+
+			if( velocityGrids && velocityGrids->readable().size() )
+			{
+				AtArray *array = ParameterAlgo::dataToArray( velocityGrids.get(), AI_TYPE_STRING );
+				AiNodeSetArray( node, g_velocityGridsArnoldString, array );
+			}
+
+			if( velocityScale != 1.0f && AiNodeEntryLookUpParameter( AiNodeGetNodeEntry( node ), g_velocityScaleArnoldString ) )
+			{
+				AiNodeSetFlt( node, g_velocityScaleArnoldString, velocityScale );
+			}
+
+			if( velocityScale > 0 )
+			{
+				AtNode *options = AiUniverseGetOptions();
+				const AtNode *arnoldCamera = static_cast<const AtNode *>( AiNodeGetPtr( options, "camera" ) );
+
+				if( arnoldCamera )
+				{
+					float shutterStart = AiNodeGetFlt( arnoldCamera, g_shutterStartArnoldString );
+					float shutterEnd = AiNodeGetFlt( arnoldCamera, g_shutterEndArnoldString );
+
+					// We're getting very lucky here:
+					//  - Arnold has automatically set options.camera the first time we made a camera
+					//  - All cameras output by Gaffer at present will have the same shutter,
+					//    so it doesn't matter if we get it from the final render camera or not.
+					AiNodeSetFlt( node, g_motionStartArnoldString, shutterStart );
+					AiNodeSetFlt( node, g_motionEndArnoldString, shutterEnd );
+				}
+			}
+
+			if( velocityFPS != 24.0f && AiNodeEntryLookUpParameter( AiNodeGetNodeEntry( node ), g_velocityFPSArnoldString ) )
```

Comment:
Again, I think we can omit the `AiNodeEntryLookUpParameter()` here. Also, do we need the `velocityFPS != 24` check? Seems like we'd be better to always set the parameter, just in case the Arnold default changes at any point. Same goes for the checks for velocityOutlierThreshold below.